spec/index.bs

```diff
@@ -15743,24 +15764,79 @@ This stage produces a <dfn dfn>Fragment</dfn> for each [=RasterizationPoint=]:
                 1. Let |vp| be |state|.{{RenderState/[[viewport]]}}.
                 1. Set |fragment|.[=Fragment/depth=] to clamp(|value|, |vp|.`minDepth`, |vp|.`maxDepth`).
+                1. Set |fragment|.[=Fragment/depthPassed=] to the result of [$compare fragment$](|fragment|.[=Fragment/destination=],
+                    |fragment|.[=Fragment/depth=], depth, |state|.{{RenderState/[[depthStencilAttachment]]}},
+                    |depthStencilDesc|?.{{GPUDepthStencilState/depthCompare}}).
         1. If `sample_mask` [=builtin=] is produced by the shader as |value|:
             1. Set |fragment|.[=Fragment/coverageMask=] to |fragment|.[=Fragment/coverageMask=] &and; |value|.
 
         Otherwise we are in [[#no-color-output]] mode, and |fragment|.[=Fragment/colors=] is empty.
     1. Return |fragment|.
 </div>
 
+<div algorithm data-timeline=queue>
+    <dfn abstract-op>compare fragment</dfn>(destination, value, aspect, attachment, compareFunc)
+
+    **Arguments:**
+
+    - |destination|: The [=FragmentDestination=].
+    - |value|: The value to be compared.
+    - |aspect|: The aspect of |attachement| to sample values from.
+    - |attachment|: The attachment to be compared against.
+    - |compareFunc|: The comparison function to use.
+
+    **Returns:** `true` if the comparison passes, or `false` otherwise
+
+    - If |attachement| is `undefined` or does not have |aspect|, return `true`.
+    - If |compareFunc| is `undefined` or {{GPUCompareFunction/"always"}}, return `true`.
+    - Let |attachmentValue| be the value of |aspect| of |attachment| at |destination|.
+    - Return `true` if comparing |value| with |attachmentValue| using |compareFunc| succeeds, and `false` otherwise.
+</div>
+
 Processing of fragments happens in parallel, while any side effects,
 such as writes into {{GPUBufferBindingType/"storage"|GPUBufferBindingType."storage"}} bindings,
 may happen in any order.
 
 ### Output Merging ### {#output-merging}
 
-<p class="note editorial"><span class=marker>Editorial note:</span> fill out this section
+Output merging is a fixed-function stage of the render [=pipeline=] that
+outputs the fragment color, depth and stencil data to be written into the render pass attachments.
 
-The depth input to this stage, if any, is clamped to the current
-{{RenderState/[[viewport]]}} depth range
-(regardless of whether the fragment shader stage writes the `frag_depth` builtin).
+<div algorithm data-timeline=queue>
+    <dfn abstract-op>process depth stencil</dfn>(fragment, descriptor, state)
+
+    **Arguments:**
+
+    - |fragment|: The [=Fragment=], produced by [[#fragment-processing]].
+    - |descriptor|: The descriptor of type {{GPURenderPipelineDescriptor}}.
+    - |state|: The active [=RenderState=].
+
+    1. Let |depthStencilDesc| be |descriptor|.{{GPURenderPipelineDescriptor/depthStencil}}.
+
+    1. If |descriptor|.{{GPURenderPipeline/[[writesDepth]]}} is `true` and |fragment|.[=Fragment/depthPassed=] is `true`:
```

Comment:
Ah! Silly mistake on my part. Let me see what feels best here. It feels a little funny that we're plumbing the pipeline descriptor and not the pipeline down through the majority of the algorithm, especially since you can get the descriptor off the pipeline.

Reply:
Resolved on #4855

It felt cleanest to simply pass in the pipeline at the top and grab the descriptor from it where needed.